Post request

anchoragewonder opened the pull request.

anchoragewonder committed:
seeting up post function

need to fix dependencies on table sql query.  add math to method for post function or table query?

anchoragewonder committed:
updating table interface

updated arguments in quiz champ table interface to be able to take a varibale and sql query based on the varibales values.

anchoragewonder committed:
add to example text

anchoragewonder committed:
Dan updates

anchoragewonder committed:
renaming sql qry to defense

anchoragewonder committed:
testing post

anchoragewonder committed:
update table interface

DanFlannel commented on League_Api.Tests/League_ApiTests.cs:
```diff
@@ -72,5 +73,15 @@ public async Task TestEmptyFunction()
+
+        [Fact]
+        public async Task TestPost()
+        {
```
Try this bit of code
```           
var function = new QuizRequestFunction();
APIGatewayProxyRequest request = new APIGatewayProxyRequest();

var body = new QuizRequestModel(2,3,2,3);
request.Body = JsonConvert.SerializeObject(body);

var context = new TestLambdaContext();
var response = await function.Execute(request, context);

output.WriteLine(Regex.Unescape(response.Body));
Assert.NotNull(response);
```

DanFlannel commented on League_Api.Tests/League_ApiTests.cs:
```diff
@@ -72,5 +73,15 @@ public async Task TestEmptyFunction()
             Assert.NotNull(response.Body);
             Assert.Equal(200, response.StatusCode);            
         }
+
```
Add this to this file

```
public League_ApiTests(ITestOutputHelper output)
        {
            this.output = output;
        }
```

anchoragewonder committed:
dan fixed it

anchoragewonder committed:
spelling is hard

anchoragewonder committed:
update package

anchoragewonder committed:
yaml description

anchoragewonder committed:
spelling

anchoragewonder committed:
yaml update

anchoragewonder committed:
added headers to lambda

anchoragewonder committed:
dont forget the bracket

anchoragewonder committed:
update yaml

anchoragewonder committed:
undo changes cors doesnt work here

anchoragewonder committed:
adjusting lambda function

ideally adding cors into the header in lamda function will pass it on into proxy integration response which is where the problem currently resides. beacuse its proxy api gateway does not give acces and problem must be fixed within the lambda function itself

anchoragewonder committed:
changes

anchoragewonder committed:
yaml updates

anchoragewonder committed:
adding cors to everything
